compiler/AST/primitive.cpp

```diff
@@ -281,8 +281,20 @@ static Type*
 returnInfoEndCount(CallExpr* call) {
   static Type* endCountType = NULL;
   if (endCountType == NULL) {
+    // Ugly hack. Basically we want to use network atomics when possible
```

Comment:
I'm ok with this going in as is, but I think you could do something like the  `_endCountAlloc(false)` thing. 

I think you could have a type after the declaration of _EndCount in ChapelBase like:

```chapel
type nonLocalEndCountType = _endCountAlloc(false).type;
```

and then search for the type `nonLocalEndCountType` by name in this loop or attach a pragma to the type and look for types with that pragma. i.e. something like 

```chapel
pragma "non local end count type"
type nonLocalEndCountType = _endCountAlloc(false).type;
```

```c
forv_Vec(TypeSymbol, ts, gTypeSymbols) {
  if (ts->hasFlag(FLAG_NON_LOCAL_END_COUNT_TYPE)) {
    ...
  }
}
```


__Edit:__

I don't remember if you can apply pragmas to types so if that doesn't work you could do:


```chapel
  pragma "non local end count type"
  proc getNonLocalEndCountType type {
    return _endCountAlloc(false).type;
  }
```

```c
returnInfoEndCount(CallExpr* call) {
  static Type* endCountType = NULL;
  if (endCountType == NULL) {
    forv_Vec(FnSymbol, fn, gFnSymbols) {
      if (fn->hasFlag(FLAG_NON_LOCAL_END_COUNT_TYPE)) {
        endCountType = fn->retType;
      }
    }
  }
  return endCountType;
}
```